crates/turborepo-scm/src/manual.rs

```diff
@@ -30,17 +30,19 @@ pub(crate) fn hash_files(
     let mut hashes = GitHashes::new();
     for file in files.into_iter() {
         let path = root_path.resolve(file.as_ref());
-        let metadata = match path.symlink_metadata() {
-            Ok(metadata) => metadata,
-            Err(e) => {
-                if allow_missing && e.is_io_error(std::io::ErrorKind::NotFound) {
-                    continue;
+        match git_like_hash_file(&path) {
+            Ok(hash) => hashes.insert(file.as_ref().to_unix(), hash),
+            Err(e) => match e {
+                Error::Io(ref io_error, _) => {
+                    if allow_missing && io_error.kind() == ErrorKind::NotFound {
+                        continue;
+                    } else {
+                        return Err(e);
+                    }
                 }
-                return Err(e.into());
-            }
+                _ => return Err(e),
+            },
         };
-        let hash = git_like_hash_file(&path, &metadata)?;
```

Comment:
Point of failure two: `fopen`. `fopen` also happens to follow symlinks, which means that our metadata call for returning size is not correlated the the actual number of bytes read in.